### ISSUE TITLE
Report validation on submit: verification statement presence only in certain cases

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-reporting#450]
#### Description:

After #405 is completed, we can amend the verification statement validation to include the following rules:

>### Verification Page Requirement 
>1. **Registration Purpose**: Users **must** complete the verification page if the registration purpose is any of the following: 
>  - OBPS Regulated Operation 
>  - Opt-in 
>  - New Entrants 
>2. **Emissions Threshold**: Users **must** complete the verification page if: 
>  - The registration purpose is **Reporting Operation**, **and** 
>  - Total attributable emissions (as shown on the Emissions Summary page) are **greater than or equal to 25,000 TCo₂e**. 
>  - 
>---

#### Acceptance Criteria:

Given
When
Then

#### Development Checklist:

- [x] In file `bc_obps/reporting/service/report_submission_service.py` refactor `def validate_report` to use  verification statement is mandatory logic in service `ReportVerificationService.get_report_needs_verification(version_id)`
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by rdromey** [comment]
@shon to write AC+Dev Checklist

**Comment by shon-button** [comment]
@patriciarussellCAS 

This should be tested with release v1.18.0 - after the refactor of conditions for when to display or not addressed in [PR 2625](https://github.com/bcgov/cas-registration/pull/2625)

[PR Test Steps](https://github.com/bcgov/cas-registration/pull/2625)
Replace `http://localhost:3000/` with `https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/`

**Comment by fviduya** [comment]
This doesn't have an Acceptance Criteria yet, but I've tested what's on the description:

I have tested this in the DEV environment, [here's](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EX91tiZqMhNErH1dVWVKHr0BH5yIjZyioJzQak1DGYqMpQ?e=9vcOcL) the test document for reference.

I have verified that Verification page is displayed on the task list under Sign-off & Submit only when:

Registration Purpose = OBPS Regulated Operation, New Entrant, Opted-In, Reporting Operation (Total attributable emissions >= 25,000 tc02e)

~I was not able to test the condition where Registration Purpose = Reporting Operation and total Emission Summary >= 25,000 tc02e _due to issue with getting into Emission Summary page_~

I also tested some scenarios where Verification page should not be visible:

Registration Purpose = Potential Reporting Operation - RESULT: Verification not showing on the task list, however, if you click Back button twice it will show the page itself, breadcrumbs also show Verification. Please check page 7 of the attached document.
^ Same as above for Registration Purpose = Reporting Operation < 25,000 tc02e

cc: @vesselak / @patriciarussellCAS

**Comment by patriciarussellCAS** [comment]
Thanks for the review. I have created ticket #505 to address the issue flagged here. otherwise this is working as intended.
